src/buildApplication/buildWorkerFile.ts

```diff
@@ -2,15 +2,17 @@ import { writeFile } from 'fs/promises';
 import { join } from 'path';
 import { build } from 'esbuild';
 import { tmpdir } from 'os';
-import { cliLog, CliOptions } from '../cli';
+import { cliLog, NodeEnv } from '../cli';
 import { NextJsConfigs } from './nextJsConfigs';
 import { MiddlewareManifestData } from './middlewareManifest';
+import { generateGlobalJs } from './buildGlobalContext';
 
 export async function buildWorkerFile(
 	{ hydratedMiddleware, hydratedFunctions }: MiddlewareManifestData,
 	vercelConfig: VercelConfig,
 	nextJsConfigs: NextJsConfigs,
-	experimentalMinify: Pick<CliOptions, 'experimentalMinify'>
+	experimentalMinify: boolean,
```

Comment:
I was passing an object here and treat is as a boolean, so it was always truthy... sorry about that :facepalm: